CMakeLists.txt

```diff
@@ -823,9 +625,22 @@ if(BOUT_TESTS)
   add_dependencies(check check-unit-tests check-integrated-tests check-mms-tests)
 endif()
 
+##################################################
+# Examples
+
 option(BOUT_BUILD_EXAMPLES "Build the examples" OFF)
 if(BOUT_BUILD_EXAMPLES)
   add_subdirectory(examples)
+else()
+  # Only build some examples (for testing)
+  add_subdirectory(examples/laplacexy/simple)
+  add_subdirectory(examples/hasegawa-wakatani-3d)
+  add_subdirectory(examples/elm-pb)
+  add_subdirectory(examples/elm-pb-outerloop)
+
+  if(BOUT_HAS_HYPRE)
+   add_subdirectory(examples/laplacexy/simple-hypre)
+  endif ()
```

Comment:
Yes I think so. What is the best way of selectively compiling just some examples, or the one that someone wants to compile? Should a user be able to specify a list of examples to compile?

Reply:
The `--target` argument to cmake works like specifying targets to `make: `cmake --build build --target elm-pb elm-pb-outerloop simple-hypre`

This will also rebuild the library if needed

Reply:
Thanks @ZedThree Unfortunately I don't think this works: If the subdirectory hasn't been added then cmake doesn't know about the examples targets. After configuring:
```
$ cmake --build build --target 6field-simple
make: *** No rule to make target '6field-simple'.  Stop
```
We can build by going into the examples/elm-pb and configuring while defining `CMAKE_PREFIX_PATH` .
This answer may have an answer https://stackoverflow.com/a/51179670 - passing `EXCLUDE_FROM_ALL` to the `add_subdirectory` statement? I'll give it a try

Reply:
Ah success! See PR #2429